e2e-tests/src/test/azero_to_eth.rs

```diff
@@ -1,93 +1,197 @@
+
+    let get_current_balance = || async { client.balance().await };
+    wait_for_balance_change(
+        get_current_balance,
+        target_balance,
+        Some(0.into()),
+        None,
+        config.test_args.wait_max_minutes,
+    )
+    .await
+}
+
+/// One-way `Aleph Zero` -> `Ethereum` transfer through `most`.
+/// Wraps AZERO into wAZERO, and approves the `most` contract to use the wAZERO funds.
+/// Burns the required funds in the wAZERO contract on Aleph Zero.
+/// Transfers `transfer_amount` of burned wAZERO over the bridge.
+/// Waits for the transfer to complete.
+/// It relies on all the relevant contracts being deployed on both ends and the (wAZERO_ETH:wAZERO_AZERO) pair having been added to `most`.
+#[tokio::test]
+pub async fn wazero_to_wazero() -> Result<()> {
+    let config = setup_test();
+    let test_context = config.create_test_context().await?;
+    let transfer_amount = utils::parse_ether(config.test_args.transfer_amount.clone())?.as_u128();
+    let client = Client::new(test_context);
+    let initial_balance = client.balance().await?;
+
+    info!("Initial balance: {:?}", initial_balance);
+
+    info!("Wrap native AZERO into wAZERO");
+    client.wrap_wazero(transfer_amount).await?;
+
+    info!("Approve the `most` contract to use the wAZERO funds on the Aleph chain");
+    client.approve_wazero_azero(transfer_amount).await?;
+
+    info!("Request the transfer of wAZERO to the Ethereum chain");
+    client
+        .request_wazero_transfer_azero(transfer_amount)
         .await?;
-    info!("`send_request` tx info: {:?}", send_request_info);
 
-    let get_current_balance = || async {
-        let balance_current = weth_eth
-            .method::<_, u128>("balanceOf", eth_account_address)?
-            .call()
-            .await?;
-        Ok::<u128, Error>(balance_current)
-    };
+    info!("Wait for balance change");
+    let target_balance = initial_balance
+        .wrap_wazero(transfer_amount)?
+        .bridge_wazero_azero_to_eth(transfer_amount)?;
+    info!("Target balance: {:?}", target_balance);
+
+    let get_current_balance = || async { client.balance().await };
+    wait_for_balance_change(
+        get_current_balance,
+        target_balance,
+        Some(0.into()),
+        None,
+        config.test_args.wait_max_minutes,
+    )
+    .await
+}
+
+/// One-way `Aleph Zero` -> `Ethereum` transfer through `most`.
+/// Requires a prior transaction in the other direction to have completed.
+/// This is easily done by running the test for the other direction first.
+/// Approves the `most` contract to use the wETH funds.
+/// Burns the required funds in the wETH contract on Aleph Zero.
+/// Transfers `transfer_amount` of burned wETH over the bridge, the Ethereum account receives native ETH.
+/// Waits for the transfer to complete.
+/// It relies on all the relevant contracts being deployed on both ends and the (wETH_ETH:wETH_AZERO) pair having been added to `most`.
```

Comment:
Suggestion: Separate actions from assertions in these comments. These descriptions make no differentiation between what the test does and what it expects to happen as a result. "Approves" is written in the same way as "Burns", but only one of these actions is taken by the test driver, the other is what MOST is supposed to do as a result. Or just remove these altogether, I think they carry about the same information as the body of the test anyway.

Reply:
removed the steps from docs altogether

Reply:
I believe I've managed to find a compromise.